src/algorithms/base_algorithm.h

```diff
@@ -126,6 +126,27 @@ class BaseAlgorithm {
   //! Reads and sets algorithm parameters from an appropriate Protobuf message
   virtual void read_params_from_proto(const bayesmix::AlgorithmParams &params);
 
+  void set_state_proto(std::shared_ptr<google::protobuf::Message> state) {
+    curr_state.CopyFrom(
+        google::protobuf::internal::down_cast<bayesmix::AlgorithmState &>(
+            *state.get()));
+  }
```

Comment:
const pointer or pointer to const? I never understood the utility  of const pointers!

Reply:
Why not both ( ͡° ͜ʖ ͡°)

Reply:
That wouldn't be possible because we modify the pointed object when we deserialize some stuff into it

Reply:
Alright. A `const` pointer cannot be changed, e.g. re-assigned, so I think it's fair game here. It has the same utility as placing `const` before an `int` argument -- to avoid unintentional overwriting of the argument